dev/InfoBar/TestUI/Themes/InfoBarStyles.xaml

```diff
@@ -133,48 +145,109 @@
+                                    <VisualState.Setters>
+                                        <Setter Target="Container.Visibility" Value="Collapsed"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                            </VisualStateGroup>
+                            <VisualStateGroup x:Name="InfoBarModeSize">
+                                <VisualState x:Name="InfoBarFloatingSize">
+                                    <VisualState.Setters>
+                                        <Setter Target="Container.MinWidth" Value="{ThemeResource InfoBarFloatingMinWidth}"/>
+                                        <Setter Target="Container.MaxWidth" Value="{ThemeResource InfoBarFloatingMaxWidth}"/>
+                                    </VisualState.Setters>
+                                </VisualState>
                             </VisualStateGroup>
                         </VisualStateManager.VisualStateGroups>
-                        <Grid x:Name="ContentRootGrid" >
-                            <Grid.ColumnDefinitions>
-                                <ColumnDefinition Width="1*"/>
-                                <ColumnDefinition Width="1*"/>
-                                <ColumnDefinition Width="2*"/>
-                                <ColumnDefinition Width="7*"/>
-                                <ColumnDefinition Width="1*"/>
-                                <ColumnDefinition Width="1*"/>
-
-                            </Grid.ColumnDefinitions>
-
-                            <Rectangle x:Name="StandardColor" Width="50" Grid.Column="0" HorizontalAlignment="Left" VerticalAlignment="Stretch">
+
+                        <Grid x:Name="InfoBarGrid"
+                              HorizontalAlignment="Stretch"
+                              MinHeight="{ThemeResource InfoBarMinHeight}" >
+
+                            <Rectangle x:Name="StandardColor" 
+                                       HorizontalAlignment="Stretch" 
+                                       VerticalAlignment="Stretch" 
+                                       Grid.ColumnSpan="6" 
+                                       Grid.RowSpan="3">
                                 <Rectangle.Fill>
-                                    <SolidColorBrush x:Name="StandardBrush" Color="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=(local:InfoBar.StatusColor)}"/>
+                                    <SolidColorBrush x:Name="StandardBrush" 
+                                                     Color="{Binding RelativeSource={RelativeSource TemplatedParent},Path=(local:InfoBar.StatusColor)}"/>
                                 </Rectangle.Fill>
                             </Rectangle>
-                            <Rectangle x:Name="UserColor" Width="50" Grid.Column="0" HorizontalAlignment="Left" VerticalAlignment="Stretch">
+
+                            <Rectangle x:Name="UserColor" 
```

Comment:
><Rectangle x:Name="UserColor"  [](start = 28, length = 30)

Why is this a Rectangle and not just the background of the root grid?

Reply:
The idea is the two rectangles would be collapsed/visible depending on if the user wanted to use their own StatusColor or stick with the preset severity StatusColor (If UserColor is visible, then Standard Color will be collapsed and vice versa). Collapsing/making them visible lets me ensure the user's custom StatusColor won't be overwritten.